vignettes/GOstatsHyperG.Rmd

```diff
@@ -0,0 +1,521 @@
+select probes with sufficient variability across samples to be
+informative; probes with little variability across all samples are
+inherently uninteresting.  Finally, the set of remaining probes was
+refined by ensuring that each probe maps to exactly one Entrez Gene
+identifier.  For those probes mapping to the same Entrez Gene ID, the
+probe with largest IQR was selected.
+
+Producing a set of Entrez Gene identifiers that map to a unique set of
+probes at the non-specific filtering stage is important because genes
+are mapped to GO categories using Entrez Gene IDs and we want to avoid
+double counting any GO categories.  In all, the filtering left
+`r length(featureNames(nsFiltered))` genes.
+
+## Gene selection via t-test
+
+We apply a standard $t$-test to identify a set of genes with
+differential expression between the `r subsetType` and NEG groups.
+
+```{r parametric1, echo=TRUE, results='hide', cache=TRUE}
+ttestCutoff <- 0.05
+ttests = rowttests(nsFiltered, "mol.biol")
+
+smPV = ttests$p.value < ttestCutoff
+
+pvalFiltered <- nsFiltered[smPV, ] 
+selectedEntrezIds <- unlist(mget(featureNames(pvalFiltered), 
+                                 hgu95av2ENTREZID))
+```
+
+There are `r sum(smPV)` genes with $p$-values less than `r ttestCutoff`. We do not make use of any
+$p$-value correction methods since we are interested in a relatively
+long gene list.
+
+A detail often omitted from GO association analyses is the fact that the
+$t$-test, and most similar statistics, are directional. For a given
+gene, average expression might be higher in the `r subsetType` group than in the NEG
+group, whereas for a different gene it might be the NEG group that shows
+the increased expression. By only looking at the $p$-values for the test
+statistics, the directionality is lost. The danger is that an
+association with a GO category may be found where the genes are not
+differentially expressed in the same direction. One way to tackle this
+problem is by separating the selected gene list into two lists according
+to direction and running two analyses. A more elegant approach is the
+subject of further research.
+
+## Inputs
+
+Often one wishes to perform many similar analyses using slightly
+different sets of parameters and to facilitate this pattern of usage the
+main interface to the Hypergeometric tests, `hyperGTest`, takes a single
+parameter object as its argument. This argument is an instance of class
+*GOHyperGParams*. There are also parameter classes *KEGGHyperGParams*
+and *PFAMHyperGParams* defined in the `r Biocpkg("Category")` package
+that allow for testing for association with KEGG pathways and PFAM
+protein domains, respectively.
+
+Using a parameter class instead of individual arguments makes it easier
+to organize and execute a series of related analyses. For example, one
+can create a list of *GOHyperGParams* instances and perform the
+Hypergeometric test on each using R's `lapply` function:
+
+````{verbatim}
```

Comment:
Use `{r eval=FALSE}`. The code won't be evaluated but it will be marked as R code.